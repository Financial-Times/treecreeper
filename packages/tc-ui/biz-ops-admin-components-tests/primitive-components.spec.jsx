const React = require('react');

const {
	Document,
	Paragraph,
	SystemLifecycle,
	ProductLifecycle,
	ServiceTier,
	Boolean,
	Url,
	Email,
	Default,
} = require('../../../src/templates/components/primitive-components.jsx');

const { render } = require('../../testHelpers/component');

const testComponentProperties = (Component, props) => {
	it('Display of full details', () => {
		const renderedComponent = render(<Component {...props} />);
		expect(renderedComponent).toMatchSnapshot();
	});

	Object.keys(props).forEach(key => {
		it(`Display with no ${key}`, () => {
			const propsWithoutKey = { ...props };
			delete propsWithoutKey[key];
			const renderedComponent = render(
				<Component {...propsWithoutKey} />,
			);
			expect(renderedComponent).toMatchSnapshot();
		});
	});

	it(`Display with no properties`, () => {
		const noProps = {};
		const renderedComponent = render(<Component {...noProps} />);
		expect(renderedComponent).toMatchSnapshot();
	});
};

describe('<Document />', () => {
	testComponentProperties(Document, {
		value: '#heading\n\nparagraph https://biz-ops.com',
	});
});
describe('<Paragraph />', () => {
	testComponentProperties(Paragraph, {
		value: 'some text https://biz-ops.com',
	});
});
describe('<SystemLifecycle />', () => {
	testComponentProperties(SystemLifecycle, { value: 'Production' });
});
describe('<ProductLifecycle />', () => {
	testComponentProperties(ProductLifecycle, { value: 'Incubate' });
});
describe('<ServiceTier />', () => {
	testComponentProperties(ServiceTier, { value: 'Bronze' });
});
describe('<Boolean />', () => {
	testComponentProperties(Boolean, { value: true });
	testComponentProperties(Boolean, { value: false });
});
describe('<Url />', () => {
	testComponentProperties(Url, { value: 'https://biz-ops.com' });
});
describe('<Email />', () => {
	testComponentProperties(Email, { value: 'biz@ops.com' });
});

describe('<Default />', () => {
	testComponentProperties(Default, { value: 'some text' });
});

// describe('<OptionalText />', () => {
// 	testComponentProperties(OptionalText, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<OptionalLink />', () => {
// 	testComponentProperties(OptionalLink, {
// 		label: 'LABEL',
// 		value: 'TARGET',
// 		name: 'NAME',
// 	});
// });

// describe('<Boolean />', () => {
// 	testComponentProperties(Boolean, { label: 'LABEL', value: true });

// 	it('Display of full false details', () => {
// 		const component = render(
// 			<Boolean label="LABEL" value={false} />,
// 		);
// 		expect(component).toMatchSnapshot();
// 	});

// 	it('Display of full rubbish details', () => {
// 		const component = render(
// 			<Boolean label="LABEL" value="rubbish" />,
// 		);
// 		expect(component).toMatchSnapshot();
// 	});
// });

// describe('<Link />', () => {
// 	testComponentProperties(Link, {
// 		label: 'LABEL',
// 		value: 'TARGET',
// 		name: 'NAME',
// 	});
// });

// describe('<Code />', () => {
// 	testComponentProperties(Code, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<ID />', () => {
// 	testComponentProperties(ID, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<Name />', () => {
// 	testComponentProperties(Name, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<Description />', () => {
// 	testComponentProperties(Description, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<Email />', () => {
// 	testComponentProperties(Email, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<EmailLink />', () => {
// 	testComponentProperties(EmailLink, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<Slack />', () => {
// 	testComponentProperties(Slack, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<SlackLink />', () => {
// 	testComponentProperties(SlackLink, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<Phone />', () => {
// 	testComponentProperties(Phone, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<PhoneLink />', () => {
// 	testComponentProperties(PhoneLink, { label: 'LABEL', value: 'VALUE' });
// });

// describe('<LifecycleStage />', () => {
// 	testComponentProperties(LifecycleStage, {
// 		label: 'LABEL',
// 		value: 'Production',
// 	});

// 	it('Display of full Preproduction details', () => {
// 		const component = render(
// 			<LifecycleStage label="LABEL" value="PreProduction" />,
// 		);
// 		expect(component).toMatchSnapshot();
// 	});

// 	it('Display of full rubbish details', () => {
// 		const component = render(
// 			<LifecycleStage label="LABEL" value="rubbish" />,
// 		);
// 		expect(component).toMatchSnapshot();
// 	});
// });

// describe('<PrimaryURL />', () => {
// 	testComponentProperties(PrimaryURL, { label: 'LABEL', value: 'VALUE' });
// });
