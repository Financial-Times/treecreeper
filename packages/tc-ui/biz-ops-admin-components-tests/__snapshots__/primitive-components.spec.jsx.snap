// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Boolean /> Display of full details 1`] = `<span>Yes</span>`;

exports[`<Boolean /> Display of full details 2`] = `<span>No</span>`;

exports[`<Boolean /> Display with no properties 1`] = `<span>Unknown</span>`;

exports[`<Boolean /> Display with no properties 2`] = `<span>Unknown</span>`;

exports[`<Boolean /> Display with no value 1`] = `<span>Unknown</span>`;

exports[`<Boolean /> Display with no value 2`] = `<span>Unknown</span>`;

exports[`<Default /> Display of full details 1`] = `<span>some text</span>`;

exports[`<Default /> Display with no properties 1`] = `<span></span>`;

exports[`<Default /> Display with no value 1`] = `<span></span>`;

exports[`<Document /> Display of full details 1`] = `
<section>
  <p>#heading</p>
  <p>paragraph
    <a href="https://biz-ops.com">https://biz-ops.com</a>
  </p>
</section>
`;

exports[`<Document /> Display with no properties 1`] = `<section></section>`;

exports[`<Document /> Display with no value 1`] = `<section></section>`;

exports[`<Email /> Display of full details 1`] = `<a href="mailto:biz@ops.com">biz@ops.com</a>`;

exports[`<Email /> Display with no properties 1`] = `null`;

exports[`<Email /> Display with no value 1`] = `null`;

exports[`<Paragraph /> Display of full details 1`] = `
<p>some text
  <a href="https://biz-ops.com" target="_blank" rel="noopener noreferrer">biz-ops.com</a>
</p>
`;

exports[`<Paragraph /> Display with no properties 1`] = `<p></p>`;

exports[`<Paragraph /> Display with no value 1`] = `<p></p>`;

exports[`<ProductLifecycle /> Display of full details 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--support-experimental">Incubate</span>`;

exports[`<ProductLifecycle /> Display with no properties 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>`;

exports[`<ProductLifecycle /> Display with no value 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>`;

exports[`<ServiceTier /> Display of full details 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--tier-bronze">Bronze</span>`;

exports[`<ServiceTier /> Display with no properties 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>`;

exports[`<ServiceTier /> Display with no value 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown service tier</span>`;

exports[`<SystemLifecycle /> Display of full details 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--support-active">Production</span>`;

exports[`<SystemLifecycle /> Display with no properties 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>`;

exports[`<SystemLifecycle /> Display with no value 1`] = `<span class="o-layout--unstyled-element o-labels o-labels--unknown">Unknown lifecycle stage</span>`;

exports[`<Url /> Display of full details 1`] = `<a href="https://biz-ops.com">https://biz-ops.com</a>`;

exports[`<Url /> Display with no properties 1`] = `null`;

exports[`<Url /> Display with no value 1`] = `null`;
